universal-login-provider/src/ULIFrameProvider.ts

```diff
@@ -57,11 +58,23 @@ export class ULIFrameProvider {
       case 'ul_ready':
         this.isReady.set(true);
         break;
+      case 'ul_set_notification_indicator':
+        this.isNewNotifications.set(msg.params[0]);
+        this.setNotificationsIndicator(this.isNewNotifications.get());
```

Comment:
Instead of calling this function every time the state is set, you can just subscribe to the state in `initUlButton`.
I've added a possible implementation below